bugwarrior/services/azuredevops.py

```diff
@@ -0,0 +1,209 @@
+        },
+        PARENT: {"type": "string", "label": "Azure Devops Parent Work Item Name"},
+        NAMESPACE: {"type": "string", "label": "Azure Devops Namespace"},
+    }
+    UNIQUE_KEY = (URL,)
+
+    PRIORITY_MAP = {"1": "H", "2": "M", "3": "L", "4": "L"}
+
+    def get_priority(self):
+        # Map to Priority Maps
+        pass
+
+
+    def to_taskwarrior(self):
+        return {
+            "project": self.extra['project'],
+            "priority": self.get_priority(),
+            "annotations": self.extra.get("annotations", []),
+            "entry": self.parse_date(
+                self.record.get("fields", {}).get("System.CreatedDate")
+            ),
+            "end": self.parse_date(
+                self.record.get("fields",{}).get("Microsoft.VSTS.Common.ClosedDate")
+            ),
+            self.TITLE: self.record["fields"]["System.Title"],
+            self.DESCRIPTION: format_item(
+                self.record.get("fields").get("System.Description")
+            ),
+            self.ID: self.record["id"],
+            self.URL: self.get_processed_url(self.record["_links"]["html"]["href"]),
+            self.TYPE: self.record["fields"]["System.WorkItemType"],
+            self.STATE: self.record["fields"]["System.State"],
+            self.ACTIVITY: self.record.get("fields").get("System.Activity", ""),
+            self.PRIORITY: self.record.get("fields").get(
+                "Microsoft.VSTS.Common.Priority", 2
+            ),
+            self.REMAINING_WORK: self.record.get("fields").get(
+                "Microsoft.VSTS.Scheduling.RemainingWork"
+            ),
+            self.PARENT: self.record.get("ParentTitle"),
+            self.NAMESPACE: self.extra.get("namespace"),
+        }
+
+    def get_default_description(self):
+        return self.build_default_description(
+            title=self.record["fields"]["System.Title"],
+            url=self.get_processed_url(self.record["_links"]["html"]["href"]),
+            number=self.record["id"],
+            cls=self.record["fields"]["System.WorkItemType"].lower(),
+        )
+
+
+class AzureDevopsService(IssueService):
+    ISSUE_CLASS = AzureDevopsIssue
+    CONFIG_PREFIX = "ado"
+
+    def __init__(self, *args, **kw):
+        super(AzureDevopsService, self).__init__(*args, **kw)
+        self.host = self.config.get("host", "dev.azure.com")
+        self.PAT = self.config.get("PAT")
+        self.project = self.config.get("project")
+        self.org = self.config.get("organization")
+        self.client = AzureDevopsClient(
+            pat=self.PAT, project=quote(self.project), org=quote(self.org), host=self.host
+        )
+        self.query_filter = self.config.get("wiql_filter")
+
+    def get_query(self):
+        default_query = (
+            "SELECT [System.Id] FROM workitems WHERE [System.AssignedTo] = @me"
+        )
+        if self.query_filter:
+            default_query += f"AND {self.query_filter}"
```

Comment:
As alluded to in my comment in the documentation, I'm doubtful that there's a fixed default query which makes sense for all users.

Reply:
The main constraint that I was initially running into here is that there is a cap on the max number of workitems the ADO api will return (20000). However I do agree with the sentiment, and removed the limit on the query, and added an extra error in the part that pulls the workitems.